exportReport/ExtensionExportReport.java

```diff
@@ -0,0 +1,298 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ * This file is based on the Paros code file ReportLastScan.java
+ */
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+import org.parosproxy.paros.extension.ExtensionHook;
+
+import org.zaproxy.zap.extension.exportReport.Export.ExportReport;
+import org.zaproxy.zap.view.ZapMenuItem;
+
+/*
+ * AUTHOR		:	GORAN SARENKAPA - JordanGS
+ * SPONSOR		:	RYERSON UNIVERSITY
+ * CLASS		:	ExtensionExportReport.java 
+ * DESC			:	Initializes all data and creates a hook point to the report menu for the extension.
+ * CREATED ON	:	MARCH 10TH, 2016
+ * CURRENT VER	:	V1.0
+ * SOURCE		:	https://github.com/JordanGS/workspace/tree/master/zap-extensions/src/org/zaproxy/zap/extension/exportReport
+ */
+
+/* 
+ * MODIFED BY	:	<NAME> - <GIT USER>
+ * MOD DATE		:	
+ * MOD VERSION	:	<VERSION OF PLUGIN>
+ * MOD DESC		:	
+ */
+
+public class ExtensionExportReport extends ExtensionAdaptor
+{
+
+	public static final String			NAME				= "ExtensionExportReport";
+	private ZapMenuItem					menuExportReport	= null;
+	private FrameExportReport			frameER				= null;
+
+	protected final Map<String, String>	alertTypeRisk		= new HashMap<>();
+
+	/* Nav Panel creation. */
+	private PanelSource					cardSource			= null;
+	private PanelAlertRisk				cardAlertRisk		= null;
+	private PanelAlertDetails			cardAlertDetails	= null;
+
+	/* Field Limit Constants */
+	private final int					textfieldLimit		= 30;
+	private final int					textareaLimit		= 600;
+
+	/* Relational CONSTANT for export types. */
+	private ArrayList<String>			alertSeverity		= new ArrayList<String>();
+	private ArrayList<String>			alertDetails		= new ArrayList<String>();
+	private ArrayList<String>			alertAdditional		= new ArrayList<String>();
+	private int							maxList				= 0;
+	
+	public ExtensionExportReport()
+	{
+		super();
+		initialize();
+	}
+
+	public ExtensionExportReport(String name)
+	{
+		super(name);
+	}
+
+	private void initialize()
+	{
+		this.setName("ExtensionExportReport");
+	}
+
+	@Override
+	public void hook(ExtensionHook extensionHook)
+	{
+		super.hook(extensionHook);
+		if (getView() != null)
+		{
+			extensionHook.getHookMenu().addReportMenuItem(getMenuExportReport());
+		}
+	}
+
+	@Override
+	public boolean canUnload()
+	{
+		return true;
+	}
+
+	@Override
+	public void unload()
+	{
+		super.unload();
+		if (frameER != null)
+		{
+			frameER.dispose();
+			frameER = null;
+		}
+	}
+
+	private ZapMenuItem getMenuExportReport()
+	{
+		if (menuExportReport == null)
+		{
+			menuExportReport = new ZapMenuItem("menu.report.html.generate");
```

Comment:
Was this meant to leverage `Constant.messages.getString()`? 

Also on the off chance that the core value ever changes you should probably use a string specific to your extension.